### ISSUE TITLE
[array-type] ReadonlyArray is regarded as Array

### ISSUE THREAD

**Issue by koooge** [typescript-eslint/typescript-eslint#4080]
Hi there,
ReadonlyArray is regarded as Array in `array-type` rule. Similarly, readonly T[] is regarded to be the same as T[].

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

Here is an example of `'array'`.

```JSON
{
  "rules": {
    '@typescript-eslint/array-type': ['error', {
      default: 'array',
      readonly: 'array', // 'array' | 'generic' | 'array-simple'
    }],
  }
}
```

```TS
const foo: ReadonlyArray<string> = [];
```

**Expected Result**
```
4:12  error  Array type using 'ReadonlyArray<string>' is forbidden. Use 'readonly string[]' instead  @typescript-eslint/array-type
```

**Actual Result**
```
4:12  error  Array type using 'Array<string>' is forbidden. Use 'string[]' instead  @typescript-eslint/array-type
```

**Additional Info**
#4066 

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.2.0` |
| `@typescript-eslint/parser`        | `5.2.0` |
| `TypeScript`                       | `4.4.4` |
| `ESLint`                           | `8.1.0` |
| `node`                             | `16.13.0` |

**Comment by JoshuaKGoldberg** [comment]
Ah that does make sense, thanks!
